internal/runtime/kubernetes/kubeobserver/waiter.go

```diff
@@ -42,24 +41,23 @@ func WaitForCondition[Client any](ctx context.Context, cli Client, action *tasks
 	})
 }
 
-func PrepareEvent(gvk kubeschema.GroupVersionKind, namespace, name, desc string, deployable runtime.Deployable) *orchestration.Event {
+func PrepareEvent(gvk kubeschema.GroupVersionKind, namespace, name, desc string, scope string) *orchestration.Event {
 	ev := &orchestration.Event{
 		ResourceId:          fmt.Sprintf("%s/%s", namespace, name),
 		RuntimeSpecificHelp: fmt.Sprintf("kubectl -n %s describe %s %s", namespace, strings.ToLower(gvk.Kind), name),
 		Ready:               orchestration.Event_NOT_READY,
+		Scope:               scope,
 	}
 
 	switch {
 	case kubedef.IsGVKDeployment(gvk), kubedef.IsGVKStatefulSet(gvk), kubedef.IsGVKPod(gvk):
 		ev.Category = "Servers deployed"
+	case kubedef.IsGVKService(gvk):
+		ev.Category = "Services deployed"
```

Comment:
Won't this add a considerable amount of output?

Reply:
This actually accomplishes that the output shrinks, as all services are displayed in a single block.

![image](https://user-images.githubusercontent.com/2537713/199523898-83e57e2d-1954-4582-8dab-96bbbbb3c011.png)

vs

![image](https://user-images.githubusercontent.com/2537713/199524109-d19744fc-f74c-4d3f-9ce5-8b09a7345ab9.png)

This code does not decide whether to display the services in general. That happens as soon as they emit events.

Reply:
<img width="1137" alt="Screen Shot 2022-11-02 at 16 03 11" src="https://user-images.githubusercontent.com/175449/199525032-2f5c090c-a1d0-4b28-8db8-8ea8a1fdd7f3.png">

I'm comparing with this, where there is no separate "Services deployed" block, apart from the service block at the end. Which I guess you don't subsume?

Reply:
Good catch! I mostly tested with `ns test`

`ns deploy` indeed looks messy now.

![image](https://user-images.githubusercontent.com/2537713/199531726-d60d537c-5d0c-4adb-9252-96700f7819a3.png)